server/src/hardware/decoder/decoder.cpp

```diff
@@ -482,7 +482,7 @@ void Decoder::rescheduleLatchedFunctionTimer()
 
   for(const auto& f : *functions)
   {
-    if(f->timeoutSeconds.value() > 0 && f->value == true)
+    if(f->hasTimeout() > 0 && f->value == true)
```

Comment:
`hasTimeout()` is a nice solution, but you should remove the `> 0`, that is confusing :P